packages/lib-classifier/src/components/Classifier/components/SubjectViewer/components/SubjectGroupViewer/components/SGVGridCell/SGVGridCell.js

```diff
@@ -88,13 +89,16 @@ function SGVGridCell (props) {
     setChecked(toggledValue)
 
     const annotationValue = annotation?.value?.slice() || []
-    const hasCellIndex = annotationValue.includes(index)
-    
-    if (hasCellIndex && !toggledValue) {  // Remove cell index from annotation values
-      const indexInValue = annotationValue.indexOf(index)
+    const isThisCellSelected = annotationValue.find(item => item.index === index)
+        
+    if (isThisCellSelected && !toggledValue) {  // Remove cell index from annotation values
+      const indexInValue = annotationValue.indexOf(isThisCellSelected)
       annotationValue.splice(indexInValue, 1)
-    } else if (!hasCellIndex && toggledValue) {  // Add cell index to annotation values
-      annotationValue.push(index)
+    } else if (!isThisCellSelected && toggledValue) {  // Add cell index to annotation values
+      annotationValue.push({
+        index,
+        subject: subjectId,
+      })
```

Comment:
Is this tested? I'm not seeing changes to the SVG Grid Cell tests.

Reply:
Tests added! 👍